api/ui.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	// This is required by the statik package
+	_ "github.com/9corp/9volt/statik"
+	"github.com/InVisionApp/rye"
+	"github.com/rakyll/statik/fs"
+)
+
+// This handler serves the Dashboard: index.html from the file system
+func (a *Api) uiHandler(rw http.ResponseWriter, r *http.Request) *rye.Response {
+	_, b, _, _ := runtime.Caller(0)
+	entrypoint := fmt.Sprintf("%v/../ui/dist/index.html", filepath.Dir(b))
+
+	http.ServeFile(rw, r, entrypoint)
+
+	return nil
+}
+
+// This handler serves all Dashboard public artifacts from the file system
+func (a *Api) uiDistHandler(rw http.ResponseWriter, r *http.Request) *rye.Response {
+	_, b, _, _ := runtime.Caller(0)
+	entrypoint := fmt.Sprintf("%v/../ui/dist/", filepath.Dir(b))
+
+	x := http.StripPrefix("/dist/", http.FileServer(http.Dir(entrypoint)))
+	x.ServeHTTP(rw, r)
+	return nil
+}
+
+// This handler serves all Dashboard public artifacts from a static file system in memory
+func (a *Api) uiDistStatikHandler(rw http.ResponseWriter, r *http.Request) *rye.Response {
+	statikFS, err := fs.New()
+	if err != nil {
+		return &rye.Response{
+			StatusCode: http.StatusInternalServerError,
+			Err:        err,
+		}
+	}
+
+	x := http.StripPrefix("/dist/", http.FileServer(statikFS))
+	x.ServeHTTP(rw, r)
+	return nil
+}
+
+// This handler serves the index.html from a static file system in memory
+func (a *Api) uiStatikHandler(rw http.ResponseWriter, r *http.Request) *rye.Response {
+	statikFS, err := fs.New()
+	if err != nil {
+		return &rye.Response{
+			StatusCode: http.StatusInternalServerError,
+			Err:        err,
+		}
+	}
+
+	file, err := statikFS.Open("/index.html")
```

Comment:
This was the most complex thing to figure out. Since I get a File System struct out of statik, if I want a single file, All I have to do is access it like any other file system in Go. However, there's very little documentation on the statik library, but this ends up being genius. It's so cool the way this works because ultimately, any kind of files could be compiled using Statik, not just web files or such. Any images, resources, config, etc. Statik is cool!